src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java

```diff
@@ -127,16 +138,54 @@ public long countComplaintsByExerciseId(long exerciseId) {
         }
 
         databaseComplaints.get().forEach(complaint -> {
-            String submissorName = principal.getName();
-            User assessor = complaint.getResult().getAssessor();
+            if (complaint.getComplaintType() != ComplaintType.MORE_FEEDBACK) {
+                String submissorName = principal.getName();
+                User assessor = complaint.getResult().getAssessor();
+
+                if (!assessor.getLogin().equals(submissorName)) {
+                    // Remove data about the student
+                    complaint.getResult().getParticipation().setStudent(null);
+                    complaint.setStudent(null);
+                    complaint.setResultBeforeComplaint(null);
+
+                    responseComplaints.add(complaint);
+                }
+            }
+        });
+
+        return responseComplaints;
+    }
+
+    /**
+     * Given an exercise id, retrieve more feedback requests related to whoever is calling the method. Useful for creating a list of more feedback requests a tutor can review.
+     *
+     * @param exerciseId - the id of the exercise we are interested in
+     * @param principal  - the callee
+     * @return a list of complaints
+     */
+    @Transactional(readOnly = true)
+    public List<Complaint> getMyMoreFeedbackRequests(long exerciseId, Principal principal) {
+        List<Complaint> responseComplaints = new ArrayList<>();
 
-            if (!assessor.getLogin().equals(submissorName)) {
-                // Remove data about the student
-                complaint.getResult().getParticipation().setStudent(null);
-                complaint.setStudent(null);
-                complaint.setResultBeforeComplaint(null);
+        Optional<List<Complaint>> databaseComplaints = complaintRepository.findByResult_Participation_Exercise_IdWithEagerSubmissionAndEagerAssessor(exerciseId);
+
+        if (!databaseComplaints.isPresent()) {
+            return responseComplaints;
+        }
 
-                responseComplaints.add(complaint);
+        databaseComplaints.get().forEach(complaint -> {
```

Comment:
see my comment above. Information hiding should NEVER be handled in the service class and should ALWAYS be handled in the resource class!!! Please change it